enhancements/etcd/disaster-recovery-with-ceo.md

```diff
@@ -19,86 +19,154 @@ see-also:
+1. Restore single master control-plane
+
+    1. Stop all the static pods on the recovery node by moving the manifests out of kubelet manifest directory and move the etcd data-dir. This will stop all the etcds leading to total quorum loss and remove all of the data
+   directories, making the backup a single source of truth.
+    1. Select a master node to run the restore operation on (recovery node) and make sure etcd backup file in present
+   on the recovery node
+    1. copy the snapshot file on a temporary location and move the copy to backup directory to 
+  /var/lib/etcd-backup/snapshot.db on the recovery node
+    1. move the static pod manifests back into kubelet manifest directory and restore the static pod resources from the backup. Because the restoration is happening from a backed up data, the revision of the manifests and resources need to match the revision that the operator was on while taking the backup.
+    1. copy the restore-etcd-pod.yaml from backed up resources to /etc/kubernetes/manifests/etcd-pod.yaml. This will restore a single member control plane using the backed up snapshot.db and then start the etcd pod bringing the control plane up. 
+
+2. Get the operator to deploy static pods on the other master nodes.
+
+    If the member has been added to the cluster and has never been started before, but a data directory exists, it means that we have dirty data which must be removed (or archived).
+    
+    1. Fix the initialization process to detect an unstarted member with existing data-dir. 
+    1. Force etcd redeployment via `oc patch etcd forceredpeloymment=reason-time`
+    1. CEO will redeploy the modified pod resources with the next revision.
+    1. The existing nodes will restart the static pods in rolling fashion with new spec.
+    
+    Note that when we force redeployment, the recovery node will also restart with the newly generated normal spec and obtain data-dir from the started members.
+
+### Recovering a single unhealthy member
+
+If the cluster-etcd-operator status shows `Degraded` condition as `true` with the status of a single `unhealthy` member, then the etcd member could be in one of the three scenarios:
+
+1. The master node in question is down and expected to come back up. 
+
+    In this case, admin doesn't need to do anything.
+    
+1. The master node is up and generally healthy, but etcd is unhealthy. If the `endpoint status` and `endpoint health` also indicate that the member is not healthy or unreachable and if the `etcd` logs on the pod indicate that it has experienced catastrophic failure, or if the pod is crash looping and resolution is not possible. 
+
+    In that event, admin needs to force the member to rejoin by removing and redeploying. Go to the "forcing an existing member to rejoin" section.
+    
+1. The master node is down and not expected to come back.
+
+    In this case, "remove the etcd member that will never return" followed by "adding a new master node"
```

Comment:
I don't see anything in this doc entitled "adding a new master node"

Reply:
Not in this doc. It should be outside of this document. Perhaps different for different providers.